lightning/src/ln/functional_tests.rs

```diff
@@ -5037,33 +5037,32 @@ fn test_static_spendable_outputs_justice_tx_revoked_htlc_timeout_tx() {
 	check_added_monitors!(nodes[1], 1);
 
 	let node_txn = nodes[1].tx_broadcaster.txn_broadcasted.lock().unwrap();
-	assert_eq!(node_txn.len(), 4); // ChannelMonitor: justice tx on revoked commitment, justice tx on revoked HTLC-timeout, adjusted justice tx, ChannelManager: local commitment tx
+	assert_eq!(node_txn.len(), 3); // ChannelMonitor: bogus justics tx, justice tx on revoked outputs, ChannelManager: local commitment tx
 	// The first transaction generated is bogus - it spends both outputs of revoked_local_txn[0]
 	// including the one already spent by revoked_htlc_txn[0]. That's OK, we'll spend with valid
 	// transactions next...
-	assert_eq!(node_txn[0].input.len(), 2);
-	check_spends!(node_txn[0], revoked_local_txn[0]);
+	assert_eq!(node_txn[0].input.len(), 3);
+	check_spends!(node_txn[0], revoked_local_txn[0], revoked_htlc_txn[0]);
 
-	check_spends!(node_txn[1], chan_1.3);
+	assert_eq!(node_txn[1].input.len(), 2);
+	check_spends!(node_txn[1], revoked_local_txn[0], revoked_htlc_txn[0]);
+	assert!((node_txn[1].input[1].previous_output.txid == revoked_htlc_txn[0].txid() &&
+	         node_txn[1].input[0].previous_output      != revoked_htlc_txn[0].input[0].previous_output) ||
+	        (node_txn[1].input[0].previous_output.txid == revoked_htlc_txn[0].txid() &&
+	         node_txn[1].input[1].previous_output      != revoked_htlc_txn[0].input[0].previous_output));
```

Comment:
Similarly for the new compound asserts in b8d0a7f8f5b5580981f3e78f5fd10e2ac688087d?